packages/sdk/tests/contracts/deployment.test.ts

```diff
@@ -0,0 +1,286 @@
+import { createAdapters, TEST_ADDRESS } from '../setup'
+import { ethers as ethersV5 } from 'ethers-v5'
+import * as ethersV6 from 'ethers-v6'
+import { setGlobalAdapter } from '@cowprotocol/sdk-common'
+import {
+  ContractsTs,
+  CONTRACT_NAMES,
+  SALT,
+  DEPLOYER_CONTRACT,
+  deterministicDeploymentAddress,
+  proxyInterface,
+  implementationAddress,
+  ownerAddress,
+} from '@cowprotocol/sdk-contracts-ts'
+import { concat, getCreate2Address, Hex, keccak256 } from 'viem'
+
+describe('Deployment and Proxy', () => {
+  let adapters: ReturnType<typeof createAdapters>
+  let contracts: {
+    ethersV5Contracts: ContractsTs
+    ethersV6Contracts: ContractsTs
+    viemContracts: ContractsTs
+  }
+
+  beforeAll(() => {
+    adapters = createAdapters()
+    contracts = {
+      ethersV5Contracts: new ContractsTs(adapters.ethersV5Adapter),
+      ethersV6Contracts: new ContractsTs(adapters.ethersV6Adapter),
+      viemContracts: new ContractsTs(adapters.viemAdapter),
+    }
+    concat(['0x1234', '0x4567'])
+  })
+
+  describe('deterministicDeploymentAddress', () => {
+    test('should compute deployment addresses consistently across different adapters', () => {
+      // Mock ABI and bytecode for testing
+      const testArtifact = {
+        abi: [
+          {
+            inputs: [
+              { name: 'authenticator', type: 'address' },
+              { name: 'allowListConfig', type: 'address' },
+            ],
+            stateMutability: 'nonpayable',
+            type: 'constructor',
+          },
+        ],
+        bytecode: '0x608060405234801561001057600080fd5b506040516102c73803806102c78339',
+        contractName: CONTRACT_NAMES.settlement,
+      }
+
+      // Deploy arguments for the settlement contract
+      const deployArgs = [
+        '0x9008D19f58AAbD9eD0D60971565AA8510560ab41', // authenticator
+        '0x1234567890123456789012345678901234567890', // allowListConfig
+      ] as [string, string]
+
+      // Compute deployment address with each adapter
+      setGlobalAdapter(adapters.ethersV5Adapter)
+      const ethersV5Address = deterministicDeploymentAddress(testArtifact, deployArgs)
+
+      setGlobalAdapter(adapters.ethersV6Adapter)
+      const ethersV6Address = deterministicDeploymentAddress(testArtifact, deployArgs)
+
+      setGlobalAdapter(adapters.viemAdapter)
+      const viemAddress = deterministicDeploymentAddress(testArtifact, deployArgs)
```

Comment:
should: make this more modular so it is easier to add one new adapter on the future?